stdlib/Random/src/RNGs.jl

```diff
@@ -316,12 +316,22 @@ function hash_seed(seed::Union{AbstractArray{UInt32}, AbstractArray{UInt64}})
     SHA.digest!(ctx)
 end
 
+function hash_seed(str::AbstractString)
+    ctx = SHA.SHA2_256_CTX()
+    for chr in str
+        SHA.update!(ctx, reinterpret(NTuple{4, UInt8}, UInt32(chr)))
```

Comment:
`UInt32(chr)` may fail if the string is not valid UTF-8. This could use the `codeunits(String(str))` instead of the Unicode codepoints to avoid this, like is done in SHA.jl
https://github.com/JuliaCrypto/SHA.jl/blob/e1af7dd0863dee14a83550faf4b6e08971993ce8/src/SHA.jl#L104-L105

Reply:
Yes that's a drawback of this approach. But I didn't want to use `codeunits` because it doesn't yield always the same result for different string types, e.g. 
```julia
julia> using Strs

julia> codeunits(LatinStr("é"))
1-element Base.CodeUnits{UInt8, LatinStr}:
 0xe9

julia> codeunits("é")
2-element Base.CodeUnits{UInt8, String}:
 0xc3
 0xa9
```
I thought rejecting invalid strings was not a big deal, and maybe useful even. But I could as well just replace `UInt32(chr)` by `chr` above, i.e. reinterpret `chr` directly as `NTuple{4, UInt8}`.

Reply:
Yeah, with `codeunits` you can convert the `AbstractString` to a `String` to ensure it is UTF-8 encoded. With reinterpreting `chr` directly, you might need to convert to `Char` because a string might have a different subtype of `AbstractChar` as its characters.

Reply:
Oh yeah I missed your conversion to `String`! Yeah that's definitely a possibility, I initially thought this could be unnecessary overhead, but it ends up being actually more efficient by directly calling `SHA.update!` on `codeunits(String(str))`, thank you :)